### ISSUE TITLE
[BUG] k-medoids intermittent fail

### ISSUE THREAD

**Issue by TonyBagnall** [aeon-toolkit/aeon#672]
### Describe the bug

occasionally, on small problems, k-medoids crashes with the trace below. We think somehow its forming empty clusters, but not sure at all why as yet. THis can be mitigated through seeding on examples, but that does not solve the problem

https://github.com/aeon-toolkit/aeon/actions/runs/5902144207/job/16009605286

### Steps/Code to reproduce the bug

this cell generated the fail on main

```python
X, y = load_unit_test()
X_train, X_test, y_train, y_test = train_test_split(X, y)
k_medoids = TimeSeriesKMedoids(
    n_clusters=2,  # Number of desired centers
    init_algorithm="random",  # Center initialisation technique
    max_iter=10,  # Maximum number of iterations for refinement on training set
    distance="msm",  # Distance metric to use
    random_state=1,
    method="alternate",
)

k_medoids.fit(X_train)
s5 = k_medoids.score(X_test, y_test)
plot_cluster_algorithm(k_medoids, X_test, k_medoids.n_clusters)
```

### Expected results

so it should not be possible for k-medoids to have an empty cluster by definition! If that is indeed the problem herei it should not happen

### Actual results

```python-traceback

---------------------------------------------------------------------------
IndexError                                Traceback (most recent call last)
Cell In[12], line 10
      1 k_medoids = TimeSeriesKMedoids(
      2     n_clusters=2,  # Number of desired centers
      3     init_algorithm="random",  # Center initialisation technique
   (...)
      7     method="alternate",
      8 )
---> 10 k_medoids.fit(X_train)
     11 s5 = k_medoids.score(X_test, y_test)
     12 plot_cluster_algorithm(k_medoids, X_test, k_medoids.n_clusters)

File /opt/hostedtoolcache/Python/3.10.12/x64/lib/python3.10/site-packages/aeon/clustering/base.py:78, in BaseClusterer.fit(self, X, y)
     73         raise AttributeError(
     74             "self.n_jobs must be set if capability:multithreading is True"
     75         )
     77 start = int(round(time.time() * 1000))
---> 78 self._fit(X)
     79 self.fit_time_ = int(round(time.time() * 1000)) - start
     80 self._is_fitted = True

File /opt/hostedtoolcache/Python/3.10.12/x64/lib/python3.10/site-packages/aeon/clustering/k_medoids.py:193, in TimeSeriesKMedoids._fit(self, X, y)
    190 best_iters = self.max_iter
    192 for _ in range(self.n_init):
--> 193     labels, centers, inertia, n_iters = self._fit_method(X)
    194     if inertia < best_inertia:
    195         best_centers = centers

File /opt/hostedtoolcache/Python/3.10.12/x64/lib/python3.10/site-packages/aeon/clustering/k_medoids.py:413, in TimeSeriesKMedoids._alternate_fit(self, X)
    410     break
    411 old_indexes = indexes
--> 413 cluster_center_indexes = self._compute_new_cluster_centers(X,indexes)
    415 if self.verbose is True:
    416     print(f"Iteration {i}, inertia {inertia}.")  # noqa: T001, T201

File /opt/hostedtoolcache/Python/3.10.12/x64/lib/python3.10/site-packages/aeon/clustering/k_medoids.py:228, in TimeSeriesKMedoids._compute_new_cluster_centers(self, X, assignment_indexes)
    226 for i in range(self.n_clusters):
    227     curr_indexes = np.where(assignment_indexes == i)[0]
--> 228     new_center_indexes.append(self._compute_medoids(X,curr_indexes))
    229 return np.array(new_center_indexes)

File /opt/hostedtoolcache/Python/3.10.12/x64/lib/python3.10/site-packages/aeon/clustering/k_medoids.py:261, in TimeSeriesKMedoids._compute_medoids(self, X, indexes)
    259 def _compute_medoids(self, X: np.ndarray, indexes: np.ndarray):
    260     distance_matrix = self._compute_pairwise(X, indexes, indexes)
--> 261     return indexes[np.argmin(sum(distance_matrix))]

IndexError: index 0 is out of bounds for axis 0 with size 0
```

**Comment by TonyBagnall** [comment]
I've just run this 20 odd times, cant reproduce it 

```python
from sklearn.model_selection import train_test_split

from aeon.clustering.k_means import TimeSeriesKMeans
from aeon.clustering.k_medoids import TimeSeriesKMedoids
from aeon.clustering.utils.plotting._plot_partitions import plot_cluster_algorithm
from aeon.datasets import load_unit_test

X, y = load_unit_test()
X_train, X_test, y_train, y_test = train_test_split(X, y)
print(X_train.shape, y_train.shape, X_test.shape, y_test.shape)

k_means = TimeSeriesKMeans(
    n_clusters=2,  # Number of desired centers
    init_algorithm="forgy",  # initialisation technique: random, forgy or kmeans++
    max_iter=10,  # Maximum number of iterations for refinement on training set
    metric="dtw",  # Distance metric to use
    averaging_method="mean",  # Averaging technique to use
    random_state=1,  # Makes deterministic
)

k_means.fit(X_train)
plot_cluster_algorithm(k_means, X_test, k_means.n_clusters)
```

**Comment by TonyBagnall** [comment]
fixed by #672
